### PR TITLE
[ Xedra Evolved ] Caves and river content

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Mods "Cave and river content for XE"

#### Purpose of change
Adds some cave and river content for Xedra Evolved.   Vehicle is based on https://en.wikipedia.org/wiki/Lenco_BearCat

#### Describe the solution
Might require #57778 .  Adds some additional locations and monsters, NPCs and items.   If I get really productive I'll add some missions, dialogue and lore.  If I go insane I may start adding mutations.  

#### Describe alternatives you've considered
None

#### Testing
Tested and it seems to all work.

### CODE REVIEW

**Comment by Maleclypse** [comment]
@Ramza13 Sorry to bother, I'm wanting to add amutation that has an EOC that dumps `"vitamins": [ [ "mutagen", 750, 850 ] ]` into the NPC randomly every few days and then it deactivates itself and another mutation a random number of days maybe 9-12 after crossing a mutation category Threshold.   Would you be able to help me get started on that?

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 1]
```diff
@@ -20,14 +20,15 @@
     "melee_skill": 3,
     "melee_dice": 1,
     "melee_dice_sides": 3,
-    "melee_damage": [ { "damage_type": "cut", "amount": 2 } ],
+    "melee_cut": 2,
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:23]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 2 } ],
```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 2]
```diff
@@ -66,7 +67,7 @@
     "speed": 110,
     "melee_dice": 2,
     "melee_dice_sides": 5,
-    "melee_damage": [ { "damage_type": "cut", "amount": 8 } ],
+    "melee_cut": 8,
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:70]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 8 } ],
```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 3]
```diff
@@ -87,7 +88,7 @@
     "melee_dice": 4,
     "melee_dice_sides": 5,
     "dodge": 4,
-    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
+    "melee_cut": 12,
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:91]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 4]
```diff
@@ -109,7 +110,7 @@
...
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:113]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 5]
```diff
@@ -137,7 +138,7 @@
     "melee_skill": 7,
     "melee_dice": 8,
     "melee_dice_sides": 3,
-    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
+    "melee_cut": 12,
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:141]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 12 } ],
```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 6]
```diff
@@ -176,7 +177,7 @@
     "melee_skill": 3,
     "melee_dice": 2,
     "melee_dice_sides": 4,
-    "melee_damage": [ { "damage_type": "cut", "amount": 6 } ],
+    "melee_cut": 6,
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:180]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 6 } ],
```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 7]
```diff
@@ -224,7 +225,7 @@
     "melee_skill": 3,
     "melee_dice": 1,
     "melee_dice_sides": 6,
-    "melee_damage": [ { "damage_type": "cut", "amount": 0 } ],
+    "melee_cut": 0,
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:228]:
```suggestion

```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 8]
```diff
@@ -258,7 +259,7 @@
     "melee_skill": 2,
     "melee_dice": 2,
     "melee_dice_sides": 4,
-    "melee_damage": [ { "damage_type": "cut", "amount": 2 } ],
+    "melee_cut": 2,
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:262]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 2 } ],
```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 9]
```diff
@@ -321,7 +322,7 @@
     "melee_skill": 6,
     "melee_dice": 2,
     "melee_dice_sides": 4,
-    "melee_damage": [ { "damage_type": "cut", "amount": 3 } ],
+    "melee_cut": 3,
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:325]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 3 } ],
```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 10]
```diff
@@ -354,7 +355,7 @@
     "melee_skill": 4,
     "melee_dice": 1,
     "melee_dice_sides": 6,
-    "melee_damage": [ { "damage_type": "cut", "amount": 6 } ],
+    "melee_cut": 6,
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:358]:
```suggestion
    "melee_damage": [ { "damage_type": "stab", "amount": 6 } ],
```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 11]
```diff
@@ -386,7 +387,7 @@
     "melee_skill": 6,
     "melee_dice": 3,
     "melee_dice_sides": 12,
-    "melee_damage": [ { "damage_type": "cut", "amount": 5 } ],
+    "melee_cut": 5,
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:390]:
```suggestion
    "melee_damage": [ { "damage_type": "stab", "amount": 5 } ],
```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 12]
```diff
@@ -370,7 +371,7 @@
     "id": "mon_xe_unicorn",
     "type": "MONSTER",
     "name": { "str": "unicorn" },
-    "description": "A hooved grazing creature with a thick, shaggy, shimmering mane of hair, a sweeping tail, and powerful-looking muscles.  Impossible to confuse with a regular horse even if you don't spot the spiraled horn coming from the head.  Rumor has it that some of the Fair Lords have ways to tame and ride this magnificent creature.",
+    "description": "A hooved grazing creature with a thick, shaggy, shimmering mane of hair, a sweeping tail, and powerful-looking muscles.  Impossible to confuse with a regular horse even if you don't spot the spiraled horn coming from the head.  Rumor has it that some of the Fair Lord's have ways to tame and ride this magnificent creature.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:374]:
```suggestion
    "description": "A hooved grazing creature with a thick, shaggy, shimmering mane of hair, a sweeping tail, and powerful-looking muscles.  Impossible to confuse with a regular horse even if you don't spot the spiraled horn coming from the head.  Rumor has it that some of the Fair Lords have ways to tame and ride this magnificent creature.",
```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 13]
```diff
@@ -438,7 +439,7 @@
     "melee_skill": 3,
     "melee_dice": 2,
     "melee_dice_sides": 6,
-    "melee_damage": [ { "damage_type": "cut", "amount": 0 } ],
+    "melee_cut": 0,
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:442]:
```suggestion

```

File: data/mods/Xedra_Evolved/monsters/changeling.json [hunk 14]
```diff
@@ -451,5 +452,34 @@
...
+    "speed": 150,
+    "material": [ "fae_flesh" ],
+    "symbol": "C",
+    "color": "white_magenta",
+    "aggression": 100,
+    "morale": 105,
+    "melee_skill": 4,
+    "melee_dice": 2,
+    "melee_dice_sides": 6,
+    "melee_cut": 2,
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/monsters/changeling.json:476]:
```suggestion
    "melee_damage": [ { "damage_type": "cut", "amount": 2} ],
```

**Comment by Ramza13** [comment]
> @Ramza13 Sorry to bother, I'm wanting to add amutation that has an EOC that dumps `"vitamins": [ [ "mutagen", 750, 850 ] ]` into the NPC randomly every few days and then it deactivates itself and another mutation a random number of days maybe 9-12 after crossing a mutation category Threshold. Would you be able to help me get started on that?

I'd be happy to help.  Sounds like a good way to help test NPC EOC's.  For clarity does this need to be an NPC?  In theory if either a player or an npc has a mutation with an EOC it should work.  The difference would be that if the NPC is outside the reality bubble its EOC's wont be run until it reenters the reality bubble.

I think something like this is a part of what you are talking about
```
{
    "type": "effect_on_condition",
    "id": "test_eoc",
    "recurrence": [ "3 days", "5 days" ],
    "condition": { "u_has_trait": "new_mutation" },
    "deactivate_condition": { "not": { "u_has_trait": "new_mutation" } },
    "effect": [
      {
        "arithmetic": [ { "u_val": "vitamin", "name": "mutagen_alpha" }, "+=", { "arithmetic": [ { "rand": 100 }, "+", { "const": 750 } ] } ]
      },
      {
        "run_eocs": {
          "id": "test_eoc_2",
          "condition": { "compare_int": [ { "u_val": "vitamin" }, ">=", { "const": 1500 } ] },
          "effect": [ { "u_lose_trait": "new_mutation" }, { "u_add_trait": "new_mutation2" } ]
        }
      }
    ]
  }
```
This will do the first part of your ask, as long as you have trait new_mutation it will give you alpha mutagen every couple of days.  Then after you have enough it will give you a new mutation and remove the existing one.  I don't think thats quite what you want, gimme some more details and I can make it better.

**Comment by Maleclypse** [comment]
> This will do the first part of your ask, as long as you have trait new_mutation it will give you alpha mutagen every couple of days. Then after you have enough it will give you a new mutation and remove the existing one. I don't think thats quite what you want, gimme some more details and I can make it better.

Cool deal, so what I'm trying to do is I've created these NPCs that the player can rescue from spider caves.  Some of them will be slowly transforming into either Troglobites or Spiders after they are rescued.  But I want them to stop transforming (gaining new mutations automatically) a random amount of time after they cross the respective Threshold so they don't just keep mutating forever.  I think that covers what I'm trying to do.

**Comment by Maleclypse** [comment]
Here's what I came up with but you'll notice length of time is clearly the wrong variable.  
```
{
    "type": "effect_on_condition",
    "id": "manifest_destiny_goblin",
    "recurrence": [ "3 days", "5 days" ],
    "condition": { "u_has_trait": "MANIFEST_DESTINY_GOBLIN" },
    "deactivate_condition": { "not": { "u_has_trait": "MANIFEST_DESTINY_GOBLIN" } },
    "effect": [
      {
        "arithmetic": [ { "u_val": "vitamin", "name": "mutagen_troglobite" }, "+=", { "arithmetic": [ { "rand": 100 }, "+", { "const": 750 } ] } ]
      },
      {
        "run_eocs": {
          "id": "test_eoc_2",
          "condition": { "compare_int": [ { "u_has_trait": "THRESH_TROGLOBITE" }, { "length of time": x } ] },
          "effect": [ { "u_lose_trait": "MANIFEST_DESTINY_GOBLIN" } ]
        }
      }
    ]
  }
```

**Comment by Maleclypse** [comment]
@Ramza13 I'm getting a plethora of errors in the mutation_eocs file.  Have I messed up some key syntax because usually when I have these kinds of cascading errors it means I've misplaced something.

**Comment by Ramza13** [comment]
This should fix your errors.  Just a few misplaced } and a wrong command or two.
```
{
    "type": "effect_on_condition",
    "id": "eoc_eggsac_host",
    "//": "Detonates the eggs every once in a while.",
    "recurrence": [ "5 h", "120 h" ],
    "condition": {
      "and": [
        { "compare_int": [ "moon", "==", { "const": 3 } ] },
        { "or": [ { "u_has_trait": "EGGSAC_SURVIVABLE" }, { "u_has_trait": "EGGSAC_FATAL" } ] }
      ]
    },
    "deactivate_condition": { "not": { "or": [ { "u_has_trait": "EGGSAC_SURVIVABLE" }, { "u_has_trait": "EGGSAC_FATAL" } ] } },
    "effect": {
      "run_eocs": {
        "id": "eoc_eggsac_mantain_or_remove_switch",
        "condition": { "u_has_trait": "EGGSAC_SURVIVABLE" },
        "effect": { "weighted_list_eocs": [ [ "eggsac_mantain", { "const": 10 } ], [ "eggsac_remove", { "u_val": "strength" } ] ] },
        "false_effect": { "weighted_list_eocs": [ [ "eggsac_mantain", { "const": 25 } ], [ "eggsac_remove", { "u_val": "strength" } ] ] }
      }
    }
  },
  {
    "type": "effect_on_condition",
    "id": "eggsac_mantain",
    "//": "Spiders erupt from your body! It seems like there are still some eggs in you.",
    "effect": {
      "run_eocs": {
        "id": "eoc_eggsac_survivable_or_fatal",
        "condition": { "u_has_trait": "EGGSAC_SURVIVABLE" },
        "effect": [
          { "npc_cast_spell": { "id": "spider_emergence_weak" } },
          { "u_message": "Spiders erupt from your body!", "type": "bad" }
        ],
        "false_effect": [
          { "npc_cast_spell": { "id": "spider_emergence_strong" } },
          { "u_message": "Spiders erupt from your body!", "type": "bad" }
        ]
      }
    }
  },
  {
    "type": "effect_on_condition",
    "id": "eggsac_remove",
    "//": "Spiders erupt from your body! It seems like those were the last eggs.",
    "effect": {
      "run_eocs": {
        "id": "eoc_eggsac_remove_survivable_or_fatal",
        "condition": { "u_has_trait": "EGGSAC_SURVIVABLE" },
        "effect": [
          { "npc_cast_spell": { "id": "spider_emergence_weak" } },
          { "u_lose_trait": "EGGSAC_SURVIVABLE" },
          { "u_message": "Spiders erupt from your body!", "type": "bad" }
        ],
        "false_effect": [
          { "npc_cast_spell": { "id": "spider_emergence_strong" } },
          { "u_lose_trait": "EGGSAC_FATAL" },
          { "u_message": "Spiders erupt from your body!", "type": "bad" }
        ]
      }
    }
  },
  {
    "type": "effect_on_condition",
    "id": "manifest_destiny_goblin",
    "recurrence": [ "3 days", "5 days" ],
    "condition": { "u_has_trait": "MANIFEST_DESTINY_GOBLIN" },
    "deactivate_condition": { "not": { "u_has_trait": "MANIFEST_DESTINY_GOBLIN" } },
    "effect": [
      {
        "arithmetic": [
          { "u_val": "vitamin", "name": "mutagen_troglobite" },
          "+=",
          { "arithmetic": [ { "rand": 100 }, "+", { "const": 750 } ] }
        ]
      },
      {
        "run_eocs": {
          "id": "manifest_destiny_goblin_complete",
          "condition": { "u_has_trait": "THRESH_TROGLOBITE" },
          "effect": [ { "u_lose_trait": "MANIFEST_DESTINY_GOBLIN" }, { "u_add_trait": "MANIFESTED_DESTINY_GOBLIN" } ]
        }
      }
    ]
  },
  {
    "type": "effect_on_condition",
    "id": "manifested_destiny_goblin",
    "recurrence": [ "3 days", "5 days" ],
    "condition": { "u_has_trait": "MANIFESTED_DESTINY_GOBLIN" },
    "deactivate_condition": { "not": { "u_has_trait": "MANIFESTED_DESTINY_GOBLIN" } },
    "effect": [
      {
        "arithmetic": [
          { "u_val": "vitamin", "name": "mutagen_troglobite" },
          "+=",
          { "arithmetic": [ { "rand": 1000 }, "+", { "const": 1750 } ] }
        ]
      },
      {
        "run_eocs": {
          "id": "destiny_goblin_removal",
          "condition": { "u_has_trait": "THRESH_TROGLOBITE" },
          "effect": [ { "u_lose_trait": "MANIFESTED_DESTINY_GOBLIN" } ]
        }
      }
    ]
  }
```

**Pull request merged**